packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -165,12 +165,14 @@ test('handleTip for commit already in log', async () => {
   await (ceramic2.repository.stateManager as any)._handleTip(streamState2, stream1.state.log[1].cid)
 
   expect(streamState2.state).toEqual(stream1.state)
-  expect(retrieveCommitSpy).toBeCalledTimes(3) // TODO(1421): This should be 2!
+  // 4 IPFS retrievals - 2 each (signed commit and linked commit) for CID of commit to be applied and CID of genesis
```

Comment:
why do we need to lookup the genesis commit if we already have it in the state?

Reply:
got it, makes sense

Reply:
Yes, the count is 4 because of `verifyLoneGenesis()`.

Is the TODO still required or just a note that it would normally be 2 but this special case causes it to be 4? The count is now not more than it would be expected to be for a typical, non-genesis case.

Reply:
I think we still expect that this can and should be 2, no?  There's no intrinsic reason why applying a new commit to existing state should require re-loading the genesis commit.

In fact, I'm *still* confused why this is happening.  I just re-read [verifyLoneGenesis](https://github.com/ceramicnetwork/js-ceramic/blob/e50898ffc87e6f556bb55ba2179b67c875abb72a/packages/core/src/state-management/state-manager.ts#L97) and it looks like it only does anything if the state only has a single commit.  So I'm still confused why applying the second commit to a stream is resulting in the genesis commit being reloaded.

Reply:
Sorry, I misspoke - the extra call is not from `verifyLoneGenesis`, it's from [applyLogToState](https://github.com/ceramicnetwork/js-ceramic/pull/1683/files#diff-efa0d2e3a8268f3351f1c2bd6a47b92f790f9286a890052ffe695890b4e0d332R276) instead.

This (i.e. an extra lookup) should only happen when there is just a genesis commit in the stream prior to the commit being applied:
```
    if (state && state.log.length === 1) {
```